structuralcodes/materials/constitutive_laws.py

```diff
@@ -776,3 +884,54 @@ def set_ultimate_strain(
                 'set_ultimate_strain requires a single value or a tuple \
                 with  two values'
             )
+
+
+CONSTITUTIVE_LAWS: t.Dict[str, ConstitutiveLaw] = {
+    'elastic': Elastic,
+    'elasticplastic': ElasticPlastic,
+    'elasticperfectlyplastic': ElasticPlastic,
+    'bilinearcompression': BilinearCompression,
+    'parabolarectangle': ParabolaRectangle,
+    'popovics': Popovics,
+    'sargin': Sargin,
+}
+
+
+def get_constitutive_laws_list() -> t.List[str]:
+    """Returns a list with valid keywords for constitutive law factory."""
+    return list(CONSTITUTIVE_LAWS.keys())
+
+
+def create_constitutive_law(
+    constitutive_law_name: str, material: Material
+) -> ConstitutiveLaw:
+    """A factory function to create the constitutive law.
+
+    Args:
+        constitutive_law_name (str): a string defining a valid constitutive law
+            type. The available keys can be get with the method
+            `get_constitutive_laws_list`
+        material (Material): The material containing the properties needed for
```

Comment:
Should we perhaps here add a short note about the required dunder methods?

Reply:
Nice idea! I tried to add some lines in 16967e2fe4a6290daab9551cb04ae7167cc69c21